File: sample-api/src/main/java/com/sample/api/order/domain/OrderEntity.java
package com.sample.api.order.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Table("tb_order")
public class OrderEntity {

    @Id
    private Long id;

    @Column("shop_cd")
    private String shopCd;

    @Column("sale_date")
    private LocalDate saleDate;

    @Column("member_id")
    private Long memberId;

    @Column("order_status")
    private String orderStatus;

    @Column("orderer_name")
    private String ordererName;

    @Column("orderer_email")
    private String ordererEmail;

    @Column("orderer_phone")
    private String ordererPhone;

    @Column("order_number")
    private String orderNumber;

    @Column("total_price")
    private BigDecimal totalPrice;

    @Column("sale_price")
    private BigDecimal salePrice;

    @Column("discount_price")
    private BigDecimal discountPrice;

    @Column("order_date")
    private LocalDateTime orderDate;

    @CreatedDate
    @Column("create_time")
    private LocalDateTime createTime;

    @LastModifiedDate
    @Column("update_time")
    private LocalDateTime updateTime;
}

File: spring-kotlin/src/main/kotlin/com/example/kosample/order/domain/OrderEntity.kt
package com.sample.api.order.domain

import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.Id
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.math.BigDecimal
import java.time.LocalDate
import java.time.LocalDateTime

@Table("tb_order")
data class OrderEntity(

    @field:Id
    val id: Long? = null,

    @field:Column("shop_cd")
    val shopCd: String,

    @field:Column("sale_date")
    val saleDate: LocalDate,

    @field:Column("member_id")
    val memberId: Long,

    @field:Column("order_status")
    val orderStatus: String,

    @field:Column("orderer_name")
    val ordererName: String,

    @field:Column("orderer_email")
    val ordererEmail: String,

    @field:Column("orderer_phone")
    val ordererPhone: String,

    @field:Column("order_number")
    val orderNumber: String,

    @field:Column("total_price")
    val totalPrice: BigDecimal,

    @field:Column("sale_price")
    val salePrice: BigDecimal,

    @field:Column("discount_price")
    val discountPrice: BigDecimal,

    @field:Column("order_date")
    val orderDate: LocalDateTime,

    // CreatedDate, LastModifiedDate는 프레임워크가 관리하므로 nullable 및 기본값 null 지정
    @field:CreatedDate
    @field:Column("create_time")
    val createTime: LocalDateTime? = null,

    @field:LastModifiedDate
    @field:Column("update_time")
    val updateTime: LocalDateTime? = null
)